src/dao/MoodDao.java

```diff
@@ -31,7 +41,41 @@ public MoodDao() {
      * executing the query
      */
     public ArrayList<Mood> getAllMoods() throws DaoException {
```

Comment:
This method needs to return an ArrayList, but currently isn't returning anything because the relevant code is inside the try block. To fix this, do the following:

1. Instantiate the ArrayList before entering the try block (just like the BufferedReader).

2. Move the return statement after the finally block.

I'll note the lines below, too.